src/features/commands/use-advanced-actions.ts

```diff
@@ -0,0 +1,203 @@
+import { differenceInSeconds, parseISO } from 'date-fns'
+import { useMemberSearch } from 'features/members-search/hooks/use-member-search'
+import { useMe } from 'features/user/hooks/use-me'
+import { useEffect, useState } from 'react'
+import { toast } from 'react-hot-toast'
+import { useHistory } from 'react-router'
+import {
+  User,
+  useSharePathMutation,
+  useUsersQuery,
+} from 'types/generated/graphql'
+import { CommandLineAction } from './command-line-hook'
+
+const handleShare = (
+  user: Omit<User, 'notifications' | 'signature'>,
+  sharePath,
+) => {
+  toast.promise(
+    sharePath({ variables: { path: location.pathname, userId: user.id } }),
+    {
+      loading: 'Sharing page',
+      success: `Page shared with ${user.fullName.split(' ')[0]}`,
+      error: 'Could not share page',
+    },
+  )
+}
+
+const getUniqueUsers = (searchValue, users, myEmail) => {
+  const name = searchValue.split('@')[searchValue.split('@').length - 1]
+
+  return (
+    users?.filter(
+      (user) =>
+        user.email !== myEmail &&
+        user.fullName.toLowerCase().includes(name.toLowerCase()),
+    ) ?? []
+  )
+}
+
+const getOnlineUsers = (searchValue, users, myEmail) => {
+  const name = searchValue.split('@')[searchValue.split('@').length - 1]
+  const now = new Date()
+
+  const uniqueUsers =
+    users?.filter(
+      (user) =>
+        user.email !== myEmail &&
+        user.fullName.toLowerCase().includes(name.toLowerCase()),
+    ) ?? []
+
+  return uniqueUsers
+    .filter((user) =>
+      user.latestPresence
+        ? differenceInSeconds(now, parseISO(user.latestPresence)) <= 10
+        : false,
+    )
+    .sort((u1, u2) =>
+      u1.fullName.toLowerCase() > u2.fullName.toLowerCase() ? 1 : -1,
+    )
+}
+
+export const useAdvancedActions = (
+  searchValue: string,
+  setSearchValue: (value: string) => void,
+  hide: () => void,
+) => {
+  const history = useHistory()
+
+  const [advanced, setAdvanced] = useState(false)
+  const [options, setOptions] = useState<
+    Array<{ label: string; onResolve: () => void }>
+  >([])
+
+  const {
+    me: { email: myEmail },
+  } = useMe()
+  const { data } = useUsersQuery()
+  const [sharePath] = useSharePathMutation()
+  const [
+    { members },
+    memberSearch,
+    { loading: membersLoading },
+  ] = useMemberSearch()
+
+  const advancedActions: CommandLineAction[] = [
+    {
+      label: 'Share path',
+      onResolve: () => {
+        setSearchValue('/share @')
+        setOptions([])
+      },
+    },
+    {
+      label: 'Go to User',
+      onResolve: () => {
+        setSearchValue('/goto @')
+        setOptions([])
+      },
+    },
+    {
+      label: 'Search Members',
+      onResolve: () => {
+        setSearchValue('/search @')
```

Comment:
In my head, the `@` is for other users, whereas in this case you can search for anything really
```suggestion
        setSearchValue('/search ')
```

Reply:
@vonElfvin 
It seems little tricky to track searching value without `@` 😓
Maybe instead of `@` use something like `:` or else 🤔